pkg/process/config/yaml_config.go

```diff
@@ -303,8 +302,13 @@ func (a *AgentConfig) initProcessDiscoveryCheck() {
 
 	// We don't need to check if the key exists since we already bound it to a default in InitConfig.
 	// We use a minimum of 10 minutes for this value.
-	a.CheckIntervals[DiscoveryCheckName] =
-		time.Duration(math.Max(float64(config.Datadog.GetDuration(key(root, "interval"))), float64(10*time.Minute)))
+	const discoveryMinInterval = 10 * time.Minute
+	if configuredInterval := config.Datadog.GetDuration(key(root, "interval")); configuredInterval >= discoveryMinInterval {
+		a.CheckIntervals[DiscoveryCheckName] = configuredInterval
+	} else {
+		a.CheckIntervals[DiscoveryCheckName] = discoveryMinInterval
+		log.Infof("Invalid interval for process discovery (<= %dm) using default value of %[1]d", discoveryMinInterval.Minutes())
+	}
```

Comment:
The code would read better if you avoid else here and more complex logic there. Effectively you are just resetting to the min interval. Disclaimer - I cannot guarantee the suggestion below compiles, apply at your own risk :) 
Other nits here:
- formatting of `time.Duration`, could you not just rely on builtin formatting?
- let's use `Warnf`
- We can define the minimal interval const at the top of the file along with other consts.
```suggestion
	const discoveryMinInterval = 10 * time.Minute
	discoveryInterval := config.Datadog.GetDuration(key(root, "interval"))
	if discoveryInterval < discoveryMinInterval {
		discoveryInterval = discoveryMinInterval
		log.Warnf("Invalid interval for process discovery (< %dm), using default value of %[1]d", discoveryMinInterval.Minutes())
	}
	a.CheckIntervals[DiscoveryCheckName] = discoveryInterval
```

Reply:
Fixed in https://github.com/DataDog/datadog-agent/pull/9283/commits/4d7f0e5ecef70cab85a27ed31e2d8b498658072e